### DISCUSSION TITLE
Is there a way for custom scalar type coercers to access (input) arguments?

### DISCUSSION THREAD

**Discussion by learnerEvermore** [graphql-java/graphql-java#2644] [Q&A]
While this won't help parsing (no input arguments on inputs themselves), it can help in situations like requesting some specific representational format.

Examples:

1. Requesting a timestamp in a certain zone or offset without requiring the scalar data itself to be fetched in a type with the zone/offset (yes, it can be done that way but it complicates things with sharing said data in layers under GraphQL).
2. Requesting an IPv6 address format - from `"2001:db8::"` via `"2001:db8:0:0:0:0:0:0"` to `"2001:0db8:0000:0000:0000:0000:0000:0000"` (all are equivalent)
3. Restricting how IP ranges can be represented, e.g. `"1.2.3.0/24"`, `"1.2.3.0-1.2.3.255"`, `"1.2.3.*"`
4. Requesting numbers to be rendered as JavaScript [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt)-compatible literals (alternative specific to this one: how can this be accomplished per-request-globally and/or with a custom coercer, for `Long` types)

**Comment by bbakerman** [comment]
Scalars dont take args - they take values and convert them.

Fields take arguments and give out values that the scalar code intreprets

```
   query {
      ipValue(fmt: FORMAT1) 
  }
  
  where
  
    type Query {
      ipValue(fmt : SomeEnum) : CustomIpScalar
``` 

The above invokes the data fetcher for field `ipValue` and it has arguments.  However the value you return     from the field is sent to the custom scalar as is - no args.

If you really want to do this - perhaps encode the desired "formatting" into the value you pass to th scalar eg pass

`format|FORMAT1|value|1.2.3.0` say

**Comment by learnerEvermore** [comment]
I am aware of that, and the possibility that I expose every such scalar in a type that will envelope that. That affects both memory and performance in this case and would love to have another option, hence the question. In other words, I already have appropriate scalar types from the product and it would be inefficient to wrap them into envelopes that would also contain what GraphQL Java sort of already has "in its hands".

I guess that the answer is that the custom scalar type coercer has no way of accessing the selection/input data for fields whose values they are coercing. If true, it would be a very useful addition.
